### PR TITLE
[FX-6047] Fix UserBadge padding in PageTopBarMenu

### PR DESCRIPTION
**Pull request by ruslan-sed**
[FX-6047]

### Description

Describe the changes and motivations for the pull request.

### How to test

- [Temploy](https://picasso.toptal.net/fx-6047-fix-pagetopbar-menu-padding)

1. Go to `PageTopBar` stories page. 
2. Go to `Page.TopBar.Menu` stories.
3. Change screen size to `<1000px` and click on avatar in the right corner of `PageTopBar`. 

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| image | image |

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Double check if `picasso-tailwind-merge` requires major update (check its `README.md`)
- ~Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)~
- ~Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core~
- ~Annotate all `props` in component with documentation~
-  ~Create `examples` for component~
- ~Ensure that deployed demo has expected results and good examples~
- ~Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).~
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

**Breaking change**

- ~codemod is created and showcased in the changeset~
- ~test alpha package of Picasso in StaffPortal~

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
4. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
5. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-6047]: https://toptal-core.atlassian.net/browse/FX-6047?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ruslan-sed** [comment]
@toptal-anvil ping reviewers

**Comment by ruslan-sed** [comment]
@toptal-anvil ping reviewers

File: packages/base/UserBadge/src/UserBadge/test.tsx [hunk 1]
```diff
@@ -98,4 +107,16 @@ describe('UserBadge', () => {
 
     expect(container).toMatchSnapshot()
   })
+
+  it('passes classes.root to the root element', () => {
```
Comment (ruslan-sed) [inline packages/base/UserBadge/src/UserBadge/test.tsx:111]: Having this test before could save some time that I spent on debugging. I think we should have more tests like this.

Comment (sashuk) [reply packages/base/UserBadge/src/UserBadge/test.tsx:111]: Agree, the `classes` is part of component API, so there should be test that makes sure that `classes.*` are passed properly. Do you plan to have it discussed / recorded somewhere?

File: cypress/component/PageTopBarMenu.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  Award16,
+  Box16,
+} from '@toptal/picasso'
+import { HAPPO_TARGETS } from '@toptal/picasso-test-utils'
+
+export enum TestIds {
+  TopBarMenu = 'page-top-bar-menu-dropdown',
+}
+
+const Example = ({ src }: { src: string | null }) => (
```
Comment (sashuk) [inline cypress/component/PageTopBarMenu.spec.tsx:18]: NIT: will it be more readable if `src` is always a string?

File: packages/base/Page/src/PageTopBarMenu/PageTopBarMenu.tsx [hunk 3]
```diff
@@ -63,7 +63,7 @@ export const PageTopBarMenu = forwardRef<HTMLDivElement, Props>(
           size='xxsmall'
           data-private={dataPrivate}
           classes={{
-            root: 'p-2 [z-index:1] bg-white relative',
+            root: 'p-[1em] [z-index:1] bg-white relative',
```
Comment (TomasSlama) [inline packages/base/Page/src/PageTopBarMenu/PageTopBarMenu.tsx:66]: why is this change required? 🤔

File: packages/base/UserBadge/src/UserBadge/test.tsx [hunk 4]
```diff
@@ -98,4 +107,16 @@ describe('UserBadge', () => {
...
+    renderUserBadge(<Typography>QA tester</Typography>, {
+      name: 'Joe Doe',
+      classes: { root: 'custom-root' },
```
Comment (sashuk) [inline packages/base/UserBadge/src/UserBadge/test.tsx:114]: NIT: `custom-root` could be moved to constant

File: packages/base/Page/src/PageTopBarMenu/PageTopBarMenu.tsx (hunk 3 above)
Comment (sashuk) [reply packages/base/Page/src/PageTopBarMenu/PageTopBarMenu.tsx:66]: I think due to https://github.com/toptal/picasso/pull/4574/files#diff-a54fc1c22dc12a9e70a058b06f149850e6dd2d8a383fbf79471010a343503847L11

**Review by sashuk — APPROVED** [review]
Tested on temploy, visual test additions looks good too!

File: packages/base/UserBadge/src/UserBadge/test.tsx (hunk 1 above)
Comment (ruslan-sed) [reply packages/base/UserBadge/src/UserBadge/test.tsx:111]: Yes, I have a note to discuss this topic on the next retro.

File: packages/base/Page/src/PageTopBarMenu/PageTopBarMenu.tsx (hunk 3 above)
Comment (ruslan-sed) [reply packages/base/Page/src/PageTopBarMenu/PageTopBarMenu.tsx:66]: Yes, Sasha is right

File: cypress/component/PageTopBarMenu.spec.tsx (hunk 2 above)
Comment (ruslan-sed) [reply cypress/component/PageTopBarMenu.spec.tsx:18]: An alternative for this is to pass to `src` prop `src` or empty string like this:
```
cy.mount(<Example src={src || ''} />)
```

In this case, we will need to do it in every test, but with the current implementation we can just pass `src`, and `Example` will handle this.

**Comment by ruslan-sed** [comment]
@toptal-anvil ping reviewers

**Comment by ruslan-sed** [comment]
@toptal-anvil ping reviewers

**Review by TomasSlama — APPROVED** [review]
